lib/Extension/LanguageServerEvaluatableExpression/Handler/EvaluatableExpressionHandler.php

```diff
@@ -0,0 +1,106 @@
+namespace Phpactor\Extension\LanguageServerEvaluatableExpression\Handler;
+
+use function Amp\call;
+use Amp\Promise;
+use Microsoft\PhpParser\Parser;
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Token;
+use Phpactor\Extension\LanguageServerBridge\Converter\PositionConverter;
+use Phpactor\Extension\LanguageServerBridge\Converter\RangeConverter;
+use Phpactor\Extension\LanguageServerEvaluatableExpression\Protocol\EvaluatableExpression;
+use Phpactor\LanguageServerProtocol\Position;
+use Phpactor\LanguageServerProtocol\ServerCapabilities;
+use Phpactor\LanguageServerProtocol\TextDocumentIdentifier;
+use Phpactor\LanguageServer\Core\Handler\CanRegisterCapabilities;
+use Phpactor\LanguageServer\Core\Handler\Handler;
+use Phpactor\LanguageServer\Core\Workspace\Workspace;
+use Phpactor\TextDocument\TextDocumentBuilder;
+use Phpactor\WorseReflection\Core\Util\NodeUtil;
+use Phpactor\TextDocument\ByteOffsetRange;
+use Phpactor\LanguageServerProtocol\Range;
+
+class EvaluatableExpressionHandler implements Handler, CanRegisterCapabilities
+{
+    public function __construct(
+        private Workspace $workspace,
+        private Parser $parser,
+    ) {
+    }
+
+    public function methods(): array
+    {
+        return [
+            'textDocument/xevaluatableExpression' => 'xevaluatableExpression',
+        ];
+    }
+
+    /**
+     * @return Promise<EvaluatableExpression|null>
+     */
+    public function xevaluatableExpression(
+        TextDocumentIdentifier $textDocument,
+        Position $position
+    ): Promise {
+        return call(function () use ($textDocument, $position) {
+            $document = $this->workspace->get($textDocument->uri);
+            $offset = PositionConverter::positionToByteOffset($position, $document->text);
+            $document = TextDocumentBuilder::create($document->text)
+                ->uri($document->uri)
+                ->language('php')
+                ->build();
+
+            $char = substr($document, $offset->toInt(), 1);
+
+            // do not provide evaluatable for whitespace
+            if (trim($char) == '') {
+                return null;
+            }
+
+            $rootNode = $this->parser->parseSourceFile((string) $document, $document->uri()?->__toString());
+            $node = $rootNode->getDescendantNodeAtPosition($offset->toInt());
+            $ret = $this->nodeToEvaluatable($node);
+            return $ret;
+        });
+    }
+
+    public function registerCapabiltiies(ServerCapabilities $capabilities): void
+    {
+        $capabilities->experimental['xevaluatableExpressionProvider'] = true;
+    }
+
+    private function nodeToEvaluatable(Node $node): ?EvaluatableExpression
+    {
+        if ($node instanceof Node\Parameter) {
+            return
+                new EvaluatableExpression(
+                    expression: $node->variableName->getText($node->getFileContents()),
+                    range: $this->byteOffsetRangeForNode($node->variableName, $node),
```

Comment:
as below maybe extract an method here describing what this does and avoid the inline assignment. to avoid the `getFileContents()` `string|null` error just cast it to a string: `getText((string)$node->getFileContents())`